### ISSUE TITLE
Issue with layer-linter on Windows

### ISSUE THREAD

**Issue by SimonBiggs** [seddonym/layer_linter#89]
* Layer Linter version: `0.12.0`
* Python version: `3.7`
* Operating System: `Windows 10`

### Description

I appear to be having issues with layer-linter on Windows.

### What I did

My `layers.yml` file looks like the following:

https://gitlab.com/pymedphys/pymedphys/blob/cd8c143ab72121ae8fa53d4797d4be9eff178292/layers.yml

Of note is the following:

```
    layers:
        - (_level3)
        - (_level2)
        - _level1
```

It appears that although `_level1` does exist within the modules I am getting errors like the following:

```
Missing layer in container 'pymedphys.coll': module pymedphys.coll._level1 does not exist.
```

### Caveat

Weirdly the error appears to occur on `Windows` but not `Linux`

See the lack of error within the following pipeline:

https://gitlab.com/pymedphys/pymedphys/-/jobs/152621905

I'll see if I can set up an AppVeyor account and see if it is replicated on there.

**Comment by seddonym** [comment]
Hi Simon,

Thanks for the bug report - I'll look into it.

**Comment by SimonBiggs** [comment]
I'll also try and get to the bottom of things on my end.

**Comment by SimonBiggs** [comment]
Heya @seddonym, I have set up our testing CI to run on Windows, macOS and Linux using azure pipelines. To get master to pass on all OSs I set the layering test to skip if the OS was Windows. The following pull request is an example where I removed that test resulting in the Windows tests to fail:

https://dev.azure.com/pymedphys/pymedphys/_build/results?buildId=306

**Comment by seddonym** [comment]
Thanks Simon - I haven't made much progress on this, mainly due to not having easy access to a Windows machine. Is there anything I can do to help you debug it?

**Comment by SimonBiggs** [comment]
I think a good first step would be to set up Multi OS testing within
layer-linter itself. Would you be okay with me setting up azure pipelines
for you? Once I have it all up and running I can transfer the project over
to you.

I might fork layer linter and iterate on my fork to set up the continuous
integration. Once I've sorted it out I'll submit a pull request with the
azure pipelines file.

That way, once it's fixed it won't regress in the future to not working
again.

Once we have the tests set up, from there I'll work on creating a fix :).

**Comment by SimonBiggs** [comment]
@seddonym here are your tests set up on azure-pipelines:

https://simonbiggs.visualstudio.com/layer_linter/_build/results?buildId=2

I've also created a pull request so it should be easy enough for you to set up azure-pipelines on your own repo.

https://github.com/seddonym/layer_linter/pull/93

Hopefully that might help you get to the bottom of the issue. I only have Linux at home so this is as far as I can go at the moment.

Cheers,

Simon

**Comment by seddonym** [comment]
Hi Simon,

Thanks for your help. See my comment on the Pull Request specifically about Azure.

It's really useful to have these build results. They immediately led me to look at:

layer_linter.dependencies.scanner.PackageScanner._module_name_from_filename

Notice a line in there that assumes that file separators are forward slashes (l. 89). My guess would be this is what's causing the issue.

Maybe you could try adjusting that to use `os.path.split` and then try running it on your azure pipeline to see if it passes?

Btw, I should let you know that I am working on a new version of layer linter that probably won't suffer from this issue, so if you would rather just wait, it's got a good chance of resolving itself. But I'd be happy to consider a separate PR (not including the Azure pipeline for now) to fix the Windows issue.

**Comment by seddonym** [comment]
This is now fixed on 0.12.2, thanks for your help!

**Comment by SimonBiggs** [comment]
Thanks David :)
